app/assets/javascripts/times.js

```diff
@@ -0,0 +1,130 @@
+/* global $ */
+
+/* Datepicker */
+$(document).on('turbolinks:load', 
+    function (){
+        $('.datepicker').datepicker({todayHighlight: true, multidate: true, format: 'yyyy-mm-dd'});
+    }
+);
+
+/* Show highlighted datepicker dates */
+$(document).on('turbolinks:load',
+    function(){
+        $('.datepicker').datepicker().on('changeDate', function(e){
+            var string = $('.datepicker').val();
+            var dateArray = string.split(",");
+            var formatted = "";
+            
+            for (var i = 0; i < dateArray.length && string.length > 0; i++){
+                var unformatted = dateArray[i];
+                var niceFormat = formatDate(unformatted);
+                var element = "#"+unformatted;
+                if ($(element).length == 0){
+                    addDiv(unformatted);
+                }
+                formatted += niceFormat + '\n';
+            }
+            
+            deleteDiv(dateArray);
+            $(".dates-chosen").text(formatted);
+        });
+    }
+);
+
+/* Add Button functionality */
+$(document).on('click', '.addbutton', function(){
+    var parent_id = $(this).parent().attr('id');
+    var element = document.getElementById(parent_id);
+    var strid = "#"+parent_id;
+    var entry = createTimeEntry(parent_id);
+    element.append(entry);
+    console.log($(strid).children());
+    console.log($(strid).children().last());
+    $(strid).children().last().append('<span><button type="button" class="deletebutton btn btn-danger btn-sm">Delete Timeslot</button></span>');
+});
+
+/* Delete Button functionality */
+$(document).on('click', '.deletebutton', function(){
+    var parent = $(this).parent().parent();
+    parent.remove();
+});
+
+
+/* Time Entry Row */
+function createTimeEntry(divId){
+    var startInput = document.createElement("input");
+    startInput.type = "time";
+    startInput.value = "12:00";
+    startInput.step = "900";
+    startInput.name = "times["+divId +"][]";
+    var endInput = document.createElement("input");
+    endInput.type = "time";
+    endInput.value = "13:00";
+    endInput.step = "900";
+    endInput.name = "times["+divId +"][]";
+    
```

Comment:
Instead of defaulting the time slot to 12 - 1PM, I think it would be better to start each time slot as the next hour from the previous hour. This way, every time new time entries are created, they will not be default be duplicate time slots.